gcp/src/main/java/org/apache/iceberg/gcp/biglake/BigLakeClient.java

```diff
@@ -0,0 +1,304 @@
+                e, "Table does not exist: %s (or permission denied)", name.getTable());
+          }
+        },
+        name.getTable());
+  }
+
+  public Table updateTableParameters(TableName name, Map<String, String> parameters, String etag) {
+    Table.Builder builder = Table.newBuilder().setName(name.toString()).setEtag(etag);
+    builder.getHiveOptionsBuilder().putAllParameters(parameters);
+    return convertException(
+        () -> {
+          try {
+            return stub.updateTable(
+                UpdateTableRequest.newBuilder()
+                    .setTable(builder)
+                    .setUpdateMask(FieldMask.newBuilder().addPaths("hive_options.parameters"))
+                    .build());
+          } catch (PermissionDeniedException e) {
+            throw new NoSuchTableException(
+                e, "Table does not exist: %s (or permission denied)", name.getTable());
+          }
+        },
+        name.getTable());
+  }
+
+  public Table renameTable(TableName name, TableName newName) {
+    return convertException(
+        () -> {
+          try {
+            return stub.renameTable(
+                RenameTableRequest.newBuilder()
+                    .setName(name.toString())
+                    .setNewName(newName.toString())
+                    .build());
+          } catch (PermissionDeniedException e) {
+            throw new NoSuchTableException(
+                e, "Table does not exist: %s (or permission denied)", name.getTable());
+          } catch (com.google.api.gax.rpc.AlreadyExistsException e) {
+            throw new AlreadyExistsException(e, "Table already exists: %s", newName.getTable());
+          }
+        },
+        name.getTable());
+  }
+
+  public Table deleteTable(TableName name) {
+    return convertException(
+        () -> {
+          try {
+            return stub.deleteTable(
+                DeleteTableRequest.newBuilder().setName(name.toString()).build());
+          } catch (PermissionDeniedException e) {
+            throw new NoSuchTableException(
+                e, "Table does not exist: %s (or permission denied)", name.getTable());
+          }
+        },
+        name.getTable());
+  }
+
+  public Iterable<Table> listTables(DatabaseName name) {
+    return convertException(
+        () ->
+            stub.listTables(ListTablesRequest.newBuilder().setParent(name.toString()).build())
+                .iterateAll(),
+        name.getDatabase());
+  }
+
+  @Override
+  public void close() {
+    stub.close();
+  }
+
+  // Converts BigLake API errors to Iceberg errors.
+  private <T> T convertException(Supplier<T> result, String resourceId) {
+    try {
+      return result.get();
+    } catch (PermissionDeniedException e) {
+      throw new NotAuthorizedException(e, "BigLake API permission denied");
+    } catch (com.google.api.gax.rpc.AlreadyExistsException e) {
+      throw new AlreadyExistsException(e, "Namespace already exists: %s", resourceId);
```

Comment:
Added a comment for clarification.

Reply:
This seems extremely brittle to me. I think it will break in the future when this code is refactored or reused by someone that didn't know. I'd prefer not to have assumptions like this baked into the code paths.